models/model.go

```diff
@@ -91,6 +92,15 @@ type Application struct {
 	UpdatedAt     string                    `json:"updated_at,omitempty"`
 }
 
+type Task struct {
```

Comment:
Unless I'm mistaken, the struct `Task` defined in https://github.com/cloudfoundry/cli/blob/main/resources/task_resource.go#L8 has all required fields and those fields are all serialized correctly. In that case, maybe we should import and use the struct instead of redefining it.

If redefinition is required for a reason, `MemoryInMB` and `DiskInMB` should be typed `uint64` instead of `int` to match cf api types.

Reply:
I think the main difference is that this unmarshals the `CreatedAt` into a `time.Time`, which allows me to do simpler/more correct `.Before` comparisons in the collector's aggregation.

I think I was just trying to save memory with the `int`s calculating that it would still give us a range to ~2 Petabytes (?), but it probably doesn't make a significant difference so can change.

Reply:
If that is the only difference, would you mind getting back on the official `resources.Task` struct as there's only a single `CreateAt.Before` call site ?

The overall idea being: overriding `resources` structs is bad since it breaks the sync with cf cli and there fore must be done only when there is no other solution.

Reply:
Tried this, turns out `resources.Task` also doesn't have the `Relationships` field, needed for aggregating by app.

Reply:
I see, therefore the resource must be redefined, indeed.